### PR TITLE
T71: Add more strongSwan Charon settings

### PR DESCRIPTION
**Pull request by unixninja92**
Added more strongswan charon settings for T71

### CODE REVIEW

File: interface-definitions/ipsec-settings.xml [hunk 1]
```diff
@@ -9,12 +9,54 @@
               <help>Global IPsec settings</help>
             </properties>
             <children>
+              <leafNode name="enable-unencrypted-mainmode-messages">
```
Comment (c-po) [inline interface-definitions/ipsec-settings.xml]: Can we probably rename this leafNode to: `unencrypted-mainmode`? CLI should be not to long. B/c everything else is stated in the help message.

File: interface-definitions/ipsec-settings.xml [hunk 2]
```diff
@@ -9,12 +9,54 @@
...
+                <properties>
+                  <help>Accept unencrypted ID and Hash messages from legacy devices</help>
+                  <valueless/>
+                </properties>
+              </leafNode>
+              <leafNode name="ikesa-table-segments">
```
Comment (c-po) [inline interface-definitions/ipsec-settings.xml]: Maybe `ike-sa-segments` is more human readable?

File: interface-definitions/ipsec-settings.xml [hunk 3]
```diff
@@ -9,12 +9,54 @@
...
+                <properties>
+                  <help>The number of segments to breakup the IKE_SA hash table into</help>
+                  <constraint>
+                    <validator name="numeric" argument="--positive"/>
+                  </constraint>
+                  <constraintErrorMessage>must be positive integer</constraintErrorMessage>
+                </properties>
+              </leafNode>
+              <leafNode name="ikesa-table-size">
```
Comment (c-po) [inline interface-definitions/ipsec-settings.xml]: Can we use `ike-sa-table-size` to make it more human readable?

File: interface-definitions/ipsec-settings.xml [hunk 4]
```diff
@@ -9,12 +9,54 @@
...
+                  <constraintErrorMessage>table size must be positive integer</constraintErrorMessage>
+                </properties>
+              </leafNode>
               <leafNode name="disable-route-autoinstall">
                 <properties>
                   <valueless/>
                   <help>Do not automatically install routes to remote networks</help>
                 </properties>
               </leafNode>
+              <leafNode name="disable-virtual-ip-autoinstall">
```
Comment (c-po) [inline interface-definitions/ipsec-settings.xml:42]: Could this be shortened to `disable-address-autoinstall` then we can use it for IPv4 and IPv6.

File: interface-definitions/ipsec-settings.xml [hunk 5]
```diff
@@ -9,12 +9,54 @@
...
+                <properties>
+                  <valueless/>
+                  <help>Do not automatically install virtual ip addresses to outbound interface</help>
+                </properties>
+              </leafNode>
+              <leafNode name="send-cisco-unity-id">
```
Comment (c-po) [inline interface-definitions/ipsec-settings.xml]: I would name it only `cisco-unity` b/c everything else is explained in the help below.

File: interface-definitions/ipsec-settings.xml [hunk 6]
```diff
@@ -9,12 +9,54 @@
...
+                <properties>
+                  <valueless/>
+                  <help>Send Cisco Unity vendor ID payload (IKEv1 only)</help>
+                </properties>
+              </leafNode>
+              <leafNode name="send-strongSwan-id">
```
Comment (c-po) [inline interface-definitions/ipsec-settings.xml]: I think this is the first node having camelCase. Can we name it `strangswan-id` instead?

File: interface-definitions/ipsec-settings.xml (hunk 1 above)
Comment (unixninja92) [reply interface-definitions/ipsec-settings.xml]: good idea

**Comment by unixninja92** [comment]
I can squash down to one or two commits if needed.

**Comment by c-po** [comment]
@unixninja92 plesae do if you don't mind.

**Comment by unixninja92** [comment]
squashed

**Review by c-po — COMMENTED** [review]
WHat about the possible shorting of other nodes in this commit? Makes everyones life a bit easier?. Later API changes are always painfull.

**Pull request closed without merging**